### DISCUSSION TITLE
Domain: Queue(s)

### DISCUSSION THREAD

**Discussion by keunes** [OpenPodcastAPI/api-specs#7] [Problem definitions & solutions]
From meeting notes:
> - We need to be able to synchronize a queue of episodes in the correct order between devices
> - We need to handle multiple queues, and have graceful handling for syncing with clients/servers that cannot handle multiple queues

**Comment by JonOfUs** [comment]
Just to add some more relevant problems/questions:

- **Queue names:** Users probably want to name their queues if they have multiple. So we also need to synchronize those queue names (and allow renaming).

- **Episodes in multiple queues:** Another question is whether users want to put the same episode into multiple queues or if it is enough to allow an episode only once in all queues and it gets removed from a queue if added to another.
It could lead to problems if clients would behave differently, so this definitely needs to be addressed.

- **Multiple vs single queues:** A way to deal with clients who don't have multiple queues could be to designate a default queue. The clients who don't support multiple queues would then only sync this queue.  
Another solution could be that the queues are concatenated (put in a row) if a client accesses the single queue endpoint. But then we would need to define a queue order and it could be difficult to merge changes if a client who supports only one queue moves episodes around. So a default queue may be a bit better (if users would understand it).
